rotkehlchen/chain/ethereum/modules/nfts.py

```diff
@@ -205,13 +253,22 @@ def get_balances(
                             'is_lp': False,
                         })
                     # Always write detected nfts in the DB to have name and address associated
-                    db_data.append((nft.token_identifier, nft.name, None, None, 0, address))  # noqa: E501
+                    db_data.append((nft.token_identifier, nft.name, None, None, False, address, False, nft.image_url, str(ZERO), collection_name))  # noqa: E501
         # save opensea data in the DB
+        fresh_nfts_identifiers = [x[0] for x in db_data]
         if len(db_data) != 0:
             with self.db.user_write() as cursor:
+                # Remove NFTs from the local cache that the user has transfered.
+                # The nfts table is not referenced by any other table so the deletion doesn't
+                # raise any error.
+                cursor.execute(
```

Comment:
Hmm what exactly are you doing here? The comment is more confusing than helping and the way the code is structured is not helping.

1. You keep all nft data you get in `db_data`. 
2. You then get all nft identifiers from that data.
3. The you delete all cached identifier data that **does not** correspond to these identifiers. So you only have that data in your DB cache.
4. You add (or ignore) into the assets table all those nft identifiers from the data.
5. For each entry in the data:
    - If there is a manual price for it you are adding or replacing it to the DB cache
    - if not ... you are only upating the image for it?  Why? What if it does not even exist in the cache?

Generally this code is terribly documented, complicated and I honestly don't get what you are trying to do here. That points to a need to change things around here.